File: 1.0/1.- Nanocrystalline Gold/HPC Code/tools/h5topng.py
# h5topng.py
#
# Usage:
#   python h5topng.py <filename>
#
# Convert an HDF5 Simulation record <filename> to a mountain of png images
#

import sys
import numpy as np
import h5py as h5
import matplotlib.pyplot as plt

def main(filename):
    try:
        fid = h5.File(filename, 'r')
    except:
        print("Error opening file, exiting.")
        return 1
    for time in fid:
        group = fid[time]
        c = group['Concentration']
        n = group['Density']
        plt.imshow(n, cmap='gray')
        plt.clim(-1, 1)
        plt.imshow(c, cmap='coolwarm', alpha=0.5)
        plt.clim(0, 1)
        plt.colorbar()

        plt.savefig(time + ".png")
        print("Write figure " + time +".png")

        plt.clf()

    return 0

if __name__ == '__main__':
    main(sys.argv[1])

File: Models/1.- Nanocrystalline Gold/HPC Code/tools/h5topng.jl
# h5topng.jl
#
# Usage:
# 	julia h5topng.jl <filename>
#
# Convert an HDF5 Simulation record  <filename> into a mountain 
# of png images

using HDF5
using PyPlot

function make_pictures(filename)
	println("Loading hdf5 datafile...")
	try
		fid = h5open(filename)
	catch 
		println("Couldn't open $filename, exiting now")
		return 1
	end

	println("Writing figures...")
	for time in names(fid)
		group = fid[time]
	
		density	= group["Density"] |> read
		concentration = group["Concentration"] |> read
	
		imshow(density, cmap="gray")
		clim(-1, 1)		

		imshow(concentration, cmap="coolwarm", alpha=0.5)	
		clim(0, 1)
		colorbar()	
		
		savefig(time*".png")
		println("Wrote figure $time.png")		

		clf()
	end

	close(fid)

	return nothing
end

make_pictures(ARGS[1])
